torchdynamo/variables/builder.py

```diff
@@ -265,6 +266,13 @@ def _wrap(self, value):
             return DataClassVariable.wrap(self, value).add_guards(
                 make_guards(GuardBuilder.TYPE_MATCH)
             )
+        elif isinstance(value, slice):
+            start = ConstantVariable(value.start)
+            stop = ConstantVariable(value.stop)
+            step = ConstantVariable(value.step)
+            return SliceVariable(
+                [start, stop, step], guards=make_guards(GuardBuilder.CONSTANT_MATCH)
```

Comment:
Is `CONSTANT_MATCH` the right type of guard? The bytecode does not change if we have `slice(0, 2, 1)` or `slice(0, 6, 2)`. So, I think `CONSTANT_MATCH` might be too strict.

I tried `FUNCTION_MATCH`, but that too was causing too many recompilations. Any suggestions?

Reply:
Yes I think CONSTANT_MATCH is correct with this handling.  It will recompile with different slices.